plugin/pkg/admission/podtolerationrestriction/admission_test.go

```diff
@@ -229,57 +229,66 @@ func TestPodAdmission(t *testing.T) {
 			testName: "added memoryPressure/DiskPressure for Guaranteed pod",
 		},
 	}
-	for _, test := range tests {
-		if test.namespaceTolerations != nil {
-			tolerationStr, err := json.Marshal(test.namespaceTolerations)
-			if err != nil {
-				t.Errorf("error in marshalling namespace tolerations %v", test.namespaceTolerations)
+	// TODO: remove for loop when support for deprecated alpha annotations removed
+	defaultTolerationsAnnotations := []string{algorithm.AnnotationDefaultTolerations, algorithm.DeprecatedAnnotationDefaultTolerations}
+	tolerationWhitelistAnnotations := []string{algorithm.AnnotationTolerationsWhitelist, algorithm.DeprecatedAnnotationTolerationsWhitelist}
+
+	for i := 0; i < len(defaultTolerationsAnnotations); i++ {
```

Comment:
again, make the deprecated annotation part of the testcase instead of looping so you can test what happens when neither, one, and both are set